### PR TITLE
Release workflow opens PR with changes

### PR DESCRIPTION
**Pull request by Srihari1192**
# Issue link

closes https://github.com/project-codeflare/instascale/issues/117

# What changes have been made

- Added the workflow step in release process to create a pull request with changes of updating go dependencies
- Included pull request creation changes from https://github.com/project-codeflare/codeflare-sdk/pull/313
- added pull request branch deletion step

# Verification steps

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: .github/workflows/instascale-release.yml
Comment (sutaakar) [inline .github/workflows/instascale-release.yml:115]: This can be done right after `Wait for PR to merge` to make sure the branch is deleted even if the release action fails for some reason.

File: .github/workflows/instascale-release.yml
Comment (sutaakar) [inline .github/workflows/instascale-release.yml:88]: Indentation is off

File: .github/workflows/instascale-release.yml
Comment (Srihari1192) [reply .github/workflows/instascale-release.yml]: Done

File: .github/workflows/instascale-release.yml
Comment (sutaakar) [inline .github/workflows/instascale-release.yml:109]: Add the branch condition also here, otherwise this command will fail in case there is no branch.

Comment (Srihari1192) [reply .github/workflows/instascale-release.yml]: Sure done

**Review by anishasthana — APPROVED** [review]

**Pull request merged**
